indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,6 +161,8 @@ public IndexTask(
     super(
         // _not_ the version, just something uniqueish
         makeId(id, ingestionSchema),
+        String.format("index_%s", makeDataSource(ingestionSchema)),
```

Comment:
Thanks for explaining that, Giam. That makes sense, I'll fix and resubmit tonight.

Do you have any input regarding the abstract class that @drcallen mentioned earlier? I originally changed it, and reverted it after previous comments. Thanks!